llvm/unittests/ADT/STLExtrasTest.cpp

```diff
@@ -1016,6 +1016,35 @@ TEST(STLExtrasTest, hasSingleElement) {
   EXPECT_FALSE(hasSingleElement(S));
 }
 
+TEST(STLExtrasTest, getSingleElement) {
+  // Note: Asserting behavior of getSingleElement cannot be tested because the
+  // program would crash.
```

Comment:
You can test this with death tests. I'd further argue that if the contract for this function is to crash on ranges with a different number of elements, we should test this. Grep for `EXPECT_DEATH` in this file for other examples.